all: code refactor remove unused headers (reduce compile-build time)

GermanAizek opened the pull request:
Code refactor commits, also reduces time spent on file preprocessing by C/C++ compilers, especially useful for contributors and users source-based distros like Gentoo.

Please don't be lazy to test PR with your Meson configuration, I might have missed or not covered everything.

As soon as I finish full PR, I'll lay out benchmarks for build speed between master branch and this PR.

It would be nice to check changes with Github Actions, but @metux is still offline.

GermanAizek committed:
xnest: refactor remove unused headers (reduce build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
xkb: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
Xi: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
xfixes: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
Xext: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
render: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
randr/record: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
present/hw/pseudoramiX: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
os/Xext: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
miext/Xext/xfixes/glx: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek commented on Xext/dpmsproc.h:
```diff
@@ -1,12 +1,8 @@
 
-#include "dixstruct.h"
-
 extern int DPMSSet(ClientPtr client, int level);
```
@mikedld, after this commits, you can suggest a separate PR that will fixes my changes, it would be more convenient.

metux replied:
Please be careful with that: we shouldn't rely on anything is included indirectly (by some other header) - better have it explicit. Only those who really aren't needed at all (ie. the .c doesn't use any definitions of that one) can and should be removed.

Let's defer this one, until we've got a complete CI that's testing all the supported build-configs on all supported platforms.

GermanAizek replied:
@metux i late squash and separate PR changes.

GermanAizek committed:
mi: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
kdrive: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
vfb: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
glx: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
glamor: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
dix: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
xfree86: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
glx/eventstr: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
others

GermanAizek committed:
test: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek committed:
xfree86: refactor remove unused headers (reduce compile-build time)

Code refactor commit, also reduces time spent on file preprocessing by compiler, especially useful for contributors and users source-based distros like Gentoo.

GermanAizek commented:
### Current results PR changes

## GCC version 14.2.0 (Debian 14.2.0-19) 

PR:
![image](https://github.com/user-attachments/assets/30d6dd60-b2bf-4bcf-9c00-d45644686b6b)

master:


## Clang version 19.1.7 (3)

metux commented:
needs rebase.
